engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -128,4 +133,35 @@ public CommandDistributionRecord getCommandDistributionRecord(
         .setIntent(persistedDistribution.getIntent())
         .setRecordValue(persistedDistribution.getCommandValue());
   }
+
+  @Override
+  public void foreachPendingDistribution(final PendingDistributionVisitor visitor) {
+    final var lastDistributionKey = new MutableLong(0);
+    final var lastPendingDistribution = new MutableReference<CommandDistributionRecord>();
+
+    pendingDistributionColumnFamily.forEach(
+        (compositeKey, nil) -> {
+          final var distributionKey = compositeKey.first().inner().getValue();
+          final var partitionId = compositeKey.second().getValue();
+
+          if (lastDistributionKey.value != distributionKey) {
```

Comment:
❓ Why do we have to keep track of the last distribution and perform this check?

Reply:
It's for performance: we may encounter the same pending distribution for separate partitions. There's no need to look up the command distribution record again in that case, but we can reuse it.

❓ Should I document this somewhere? Perhaps in a comment at that check?